reactor-core/src/main/java/reactor/core/publisher/BlockingSingleSubscriber.java

```diff
@@ -124,7 +125,8 @@ final T blockingGet(long timeout, TimeUnit unit) {
 			try {
 				if (!await(timeout, unit)) {
 					dispose();
-					throw new IllegalStateException("Timeout on blocking read for " + timeout + " " + unit);
+					String errorMessage = "Timeout on blocking read for " + timeout + " " + unit;
+					throw new IllegalStateException(errorMessage, new TimeoutException(errorMessage));
```

Comment:
https://github.com/reactor/reactor-core/blob/bbc9bd42392dd4908eefe601432db2e7fc5a5613/reactor-core/src/main/java/reactor/core/publisher/Mono.java#L1800

https://github.com/reactor/reactor-core/blob/bbc9bd42392dd4908eefe601432db2e7fc5a5613/reactor-core/src/main/java/reactor/core/publisher/Flux.java#L2786

https://github.com/reactor/reactor-core/blob/bbc9bd42392dd4908eefe601432db2e7fc5a5613/reactor-core/src/main/java/reactor/core/publisher/Flux.java#L2836

I checked that `Mono#block(timeout), Flux#blockFirst(timeout), Flux#blockLast(timeout)` use this code~!

Reply:
https://github.com/reactor/reactor-core/blob/bbc9bd42392dd4908eefe601432db2e7fc5a5613/reactor-core/src/main/java/reactor/core/publisher/Mono.java#L1787

```
// AS-IS
If the provided timeout expires, a {@link RuntimeException} is thrown.

// TO-BE
If the provided timeout expires, a {@link RuntimeException} is thrown with {@link TimeoutException} as the cause.
```

(just curiosity) should we enhance document on `Mono#block*, Flux#block*` like above?

Reply:
Yes, please.

Reply:
✅ update javadoc :)